Lib/asyncio/base_events.py

```diff
@@ -303,6 +303,8 @@ def _detach(self, transport):
             self._wakeup()
 
     def _wakeup(self):
+        if self._waiters is None:
+            return
```

Comment:
Please add a detailed comment on how a double call can happen (sort of a shorter version of your main PR message)

Reply:
Addressed in d96323752a277d48d6a173576ed62ce36bd0517b